lib/src/instances/option.dart

```diff
@@ -84,11 +228,103 @@ class Some<T> extends Option<T> {
   }
 
   @override
+  bool contains(Object element) => element == value;
+
+  @override
+  T elementAt(int index) {
+    if (index != 0) {
+      throw new RangeError.index(index, this, "index", null, 1);
+    }
+
+    return value;
+  }
+
+  @override
+  bool every(bool f(T element)) => f(value);
+
+  @override
+  IterableMonad<S> expand<S>(Iterable<S> f(T element)) =>
```

Comment:
IMHO expand should return an Option, even when the contract is from Iterable, the instance is an Option and its algebra should be closed under Options if possible.